pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/InvertedIndexReader.java

```diff
@@ -19,12 +19,20 @@
 package org.apache.pinot.core.segment.index.readers;
 
 import java.io.Closeable;
+import org.apache.pinot.core.common.Predicate;
 
 
 public interface InvertedIndexReader<T> extends Closeable {
 
   /**
    * Get the document ids for the given dictionary id.
+   * @param dictId dictionary ID
    */
   T getDocIds(int dictId);
+
+  /**
+   * Get the document ids after evaluating the given predicate
+   * @param predicate predicate
+   */
+  T getDocIds(Predicate predicate);
```

Comment:
I would recommend splitting this API into a separate interface, e.g. `TextInvertedIndexReader`, and make it another entry in the IndexContainer. You won't be able to share any implementation among these 2 APIs.

Reply:
As discussed offline, this interface might help in extending it in the future when we implement native text index and add support for complex objects. 

However, I have changed the API to getDocIds(Object value) instead of getDocIds(Predicate predicate). This makes it intuitive to use our inverted index readers for both dictionary ID based lookup and raw value (or search query) based lookup.